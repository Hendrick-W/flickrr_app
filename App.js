/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */
import 'react-native-gesture-handler';
import React from 'react';
import {
  Text,
} from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import { Provider } from 'react-redux';

import DetailPage from './src/DetailPage'
import HomePage from './src/HomePage'
import FavouritePage from './src/FavouritePage'
import store from './src/store';

const Stack = createStackNavigator();
const Tab = createBottomTabNavigator();

function MainTab () {
  return (
    <Tab.Navigator headerMode="none"
    tabBarOptions={{
      activeTintColor: "#f4511e",
      labelStyle:{
        fontSize: 12,
        fontWeight:"bold"
      }
    }}
    >
      <Tab.Screen name ="Home"
        component={HomePage}
        options = {{
          tabBarIcon: ({ focused}) => (
            <MaterialIcons name="home" color={focused? "#f4511e" : "#999999"} size={focused? 35: 25} />
          ),
          tabBarLabel: "Home Page"
        }}
      />
      <Tab.Screen name ="Favourite"
        component={FavouritePage}
        options = {{
          tabBarIcon: ({ focused}) => (
            <MaterialIcons name="favorite-border" color={focused? "#f4511e" : "#999999"} size={focused? 35: 25} />
          ),
          tabBarLabel: "My Favourite(s)"
        }}  
      />
    </Tab.Navigator>
  )
}

const App: () => React$Node = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="HomePage" component={MainTab} 
          options={{
            title: 'Flickrr',
            headerStyle: {
              backgroundColor: '#f4511e',
            },
            headerTintColor: '#fff',
            headerTitleStyle: {
              alignSelf:'center',
              fontWeight: 'bold',
            },
          }}
        />
        <Stack.Screen name="Detail" component={DetailPage} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default () => {
  return (
    <Provider store = {store}>
      <App/>
    </Provider>
  )
}
