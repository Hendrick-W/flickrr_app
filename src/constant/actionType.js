export const SEARCH_REQUEST = 'SEARCH_REQUEST';
export const SEARCH_SUCCESS = 'SEARCH_SUCCESS';
export const SEARCH_FAILURE = 'SEARCH_FAILURE';

export const ADD_TO_FAVOURITE = 'ADD_TO_FAVOURITE';
export const DELETE_ITEM_FAVOURITE = 'DELETE_ITEM_FAVOURITE';
