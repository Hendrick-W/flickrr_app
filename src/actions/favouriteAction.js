/* eslint-disable prettier/prettier */
import * as types from '../constant/actionType';

export const addFavourite = (payload) => {
  return  {
    type: types.ADD_TO_FAVOURITE,
    payload,
  };
};

export const deleteFavourite = (payload) => {
  return {
    type: types.DELETE_ITEM_FAVOURITE,
    payload,
  };
};

