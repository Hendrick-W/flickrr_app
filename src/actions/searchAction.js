/* eslint-disable prettier/prettier */
import * as types from '../constant/actionType';
import axios from 'axios';

export const searchRequest = () => ({
  type: types.SEARCH_REQUEST,
});

export const searchError = (error) => ({
  type: types.SEARCH_FAILURE,
  payload: error,
});

export const searchSuccess = (result) => ({
  type: types.SEARCH_SUCCESS,
  payload: result,
});

export const searchAction = (params) => {
  return async (dispatch) => {
    try {
      dispatch(searchRequest());
      const tags = params.tags.replace(/\s/g, ',');
      const response = await axios.get(
        `https://api.flickr.com/services/feeds/photos_public.gne?format=json&jsoncallback=?&tags=${tags}&tagmode=${params.mode}`
      );

      const data = response.data.substring(1, response.data.length - 1);
      const parse_data = JSON.parse(data);
      dispatch(
        searchSuccess({result: parse_data.items, title: parse_data.title})
      );
    } catch (error) {
      console.log('ini error', error);
      dispatch(searchError('Failed to load data'));
    }
  };
};
