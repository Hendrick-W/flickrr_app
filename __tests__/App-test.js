/* eslint-disable prettier/prettier */
/**
 * @format
 */

import 'react-native';
import {Provider} from 'react-redux';
import store from '../src/store';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axios from 'axios';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import {create, act} from 'react-test-renderer';

import * as types from '../src/constant/actionType';
import {searchAction} from '../src/actions/searchAction';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const appScreen = create(
  <Provider store={store}>
    <App />
  </Provider>
);

test('snapshot', () => {
  expect(appScreen).toMatchSnapshot();
  // expect(appScreen).toBeTruthy();
});

test('Render text input search bar component', ()=> {
  act(()=>jest.runAllTimers());
  const textInput = appScreen.root.findByProps({testID: 'searchTags'}).props;
  expect(textInput).toBeDefined();
});

test('Render button search bar component', ()=> {
  act(()=>jest.runAllTimers());
  const searchButton = appScreen.root.findByProps({testID: 'searchButton'}).props;
  expect(searchButton).toBeDefined();
});
