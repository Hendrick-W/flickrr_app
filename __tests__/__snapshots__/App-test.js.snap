// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#f4511e",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#fff",
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
              >
                Flickrr
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        pointerEvents="auto"
                        style={
                          Array [
                            Object {
                              "flex": 1,
                              "overflow": "hidden",
                            },
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                          ]
                        }
                      >
                        <View
                          collapsable={false}
                          pointerEvents="auto"
                          removeClippedSubviews={false}
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            accessibilityElementsHidden={false}
                            importantForAccessibility="auto"
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                Object {
                                  "backgroundColor": "rgb(242, 242, 242)",
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              onLayout={[Function]}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "#000",
                                    "flex": 1,
                                  },
                                  Object {
                                    "paddingBottom": 0,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "flex": 1,
                                    "minHeight": "6%",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "#fff",
                                      "flex": 1,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "borderBottomRightRadius": 20,
                                        "borderTopRightRadius": 20,
                                        "flexDirection": "row",
                                        "margin": 10,
                                        "overflow": "hidden",
                                      }
                                    }
                                  >
                                    <TextInput
                                      allowFontScaling={true}
                                      autoCapitalize="none"
                                      autoCorrect={false}
                                      onChangeText={[Function]}
                                      placeholder="Add tag(s)..."
                                      rejectResponderTermination={true}
                                      style={
                                        Object {
                                          "borderBottomLeftRadius": 20,
                                          "borderTopLeftRadius": 20,
                                          "borderWidth": 1,
                                          "flex": 3,
                                          "padding": 10,
                                          "paddingRight": 50,
                                        }
                                      }
                                      testID="searchTags"
                                      underlineColorAndroid="transparent"
                                      value=""
                                    />
                                    <View
                                      accessible={true}
                                      focusable={true}
                                      onClick={[Function]}
                                      onResponderGrant={[Function]}
                                      onResponderMove={[Function]}
                                      onResponderRelease={[Function]}
                                      onResponderTerminate={[Function]}
                                      onResponderTerminationRequest={[Function]}
                                      onStartShouldSetResponder={[Function]}
                                      style={
                                        Object {
                                          "alignItems": "center",
                                          "backgroundColor": "#d3d3d3",
                                          "flex": 1,
                                          "height": "95%",
                                          "justifyContent": "center",
                                          "opacity": 1,
                                        }
                                      }
                                      testID="searchButton"
                                    >
                                      <Text
                                        allowFontScaling={false}
                                        style={
                                          Array [
                                            Object {
                                              "color": "#fff",
                                              "fontSize": 28,
                                            },
                                            undefined,
                                            Object {
                                              "fontFamily": "anticon",
                                              "fontStyle": "normal",
                                              "fontWeight": "normal",
                                            },
                                            Object {},
                                          ]
                                        }
                                      >
                                        
                                      </Text>
                                    </View>
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "justifyContent": "space-around",
                                      }
                                    }
                                  >
                                    <View
                                      accessible={true}
                                      focusable={true}
                                      onClick={[Function]}
                                      onResponderGrant={[Function]}
                                      onResponderMove={[Function]}
                                      onResponderRelease={[Function]}
                                      onResponderTerminate={[Function]}
                                      onResponderTerminationRequest={[Function]}
                                      onStartShouldSetResponder={[Function]}
                                      style={
                                        Object {
                                          "flexDirection": "row",
                                          "opacity": 1,
                                        }
                                      }
                                    >
                                      <Text
                                        allowFontScaling={false}
                                        style={
                                          Array [
                                            Object {
                                              "color": undefined,
                                              "fontSize": 20,
                                            },
                                            undefined,
                                            Object {
                                              "fontFamily": "Material Icons",
                                              "fontStyle": "normal",
                                              "fontWeight": "normal",
                                            },
                                            Object {},
                                          ]
                                        }
                                      >
                                        
                                      </Text>
                                      <Text>
                                        Include all tags
                                      </Text>
                                    </View>
                                    <View
                                      accessible={true}
                                      focusable={true}
                                      onClick={[Function]}
                                      onResponderGrant={[Function]}
                                      onResponderMove={[Function]}
                                      onResponderRelease={[Function]}
                                      onResponderTerminate={[Function]}
                                      onResponderTerminationRequest={[Function]}
                                      onStartShouldSetResponder={[Function]}
                                      style={
                                        Object {
                                          "flexDirection": "row",
                                          "opacity": 1,
                                        }
                                      }
                                    >
                                      <Text
                                        allowFontScaling={false}
                                        style={
                                          Array [
                                            Object {
                                              "color": undefined,
                                              "fontSize": 20,
                                            },
                                            undefined,
                                            Object {
                                              "fontFamily": "Material Icons",
                                              "fontStyle": "normal",
                                              "fontWeight": "normal",
                                            },
                                            Object {},
                                          ]
                                        }
                                      >
                                        
                                      </Text>
                                      <Text>
                                        Include any tag
                                      </Text>
                                    </View>
                                  </View>
                                </View>
                              </View>
                              <View
                                style={
                                  Object {
                                    "flex": 10,
                                  }
                                }
                              >
                                <RCTScrollView
                                  data={Array []}
                                  disableVirtualization={false}
                                  getItem={[Function]}
                                  getItemCount={[Function]}
                                  horizontal={false}
                                  initialNumToRender={10}
                                  keyExtractor={[Function]}
                                  maxToRenderPerBatch={10}
                                  onContentSizeChange={[Function]}
                                  onEndReachedThreshold={2}
                                  onLayout={[Function]}
                                  onMomentumScrollEnd={[Function]}
                                  onScroll={[Function]}
                                  onScrollBeginDrag={[Function]}
                                  onScrollEndDrag={[Function]}
                                  removeClippedSubviews={false}
                                  renderItem={[Function]}
                                  scrollEventThrottle={50}
                                  stickyHeaderIndices={Array []}
                                  updateCellsBatchingPeriod={50}
                                  viewabilityConfigCallbackPairs={Array []}
                                  windowSize={21}
                                >
                                  <View />
                                </RCTScrollView>
                              </View>
                              <View>
                                <View
                                  accessibilityRole="button"
                                  accessibilityState={Object {}}
                                  accessible={true}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "opacity": 1,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {},
                                      ]
                                    }
                                  >
                                    <Text
                                      style={
                                        Array [
                                          Object {
                                            "color": "#007AFF",
                                            "fontSize": 18,
                                            "margin": 8,
                                            "textAlign": "center",
                                          },
                                        ]
                                      }
                                    >
                                      Refresh
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                    <View
                      pointerEvents="auto"
                      style={
                        Object {
                          "backgroundColor": "rgb(255, 255, 255)",
                          "borderTopColor": "rgb(216, 216, 216)",
                          "borderTopWidth": 0.5,
                          "bottom": 0,
                          "elevation": 8,
                          "height": 49,
                          "left": 0,
                          "paddingBottom": 0,
                          "paddingHorizontal": 0,
                          "position": null,
                          "right": 0,
                          "transform": Array [
                            Object {
                              "translateY": 0,
                            },
                          ],
                        }
                      }
                    >
                      <View
                        onLayout={[Function]}
                        style={
                          Object {
                            "flex": 1,
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          accessibilityLabel="Home Page, tab, 1 of 2"
                          accessibilityRole="button"
                          accessibilityState={
                            Object {
                              "selected": true,
                            }
                          }
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flex": 1,
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                              Object {
                                "flexDirection": "column",
                                "justifyContent": "flex-end",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "center",
                                    "height": "100%",
                                    "justifyContent": "center",
                                    "minWidth": 25,
                                    "position": "absolute",
                                    "width": "100%",
                                  },
                                  Object {
                                    "opacity": 1,
                                  },
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#f4511e",
                                      "fontSize": 35,
                                    },
                                    undefined,
                                    Object {
                                      "fontFamily": "Material Icons",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                              >
                                
                              </Text>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "center",
                                    "height": "100%",
                                    "justifyContent": "center",
                                    "minWidth": 25,
                                    "position": "absolute",
                                    "width": "100%",
                                  },
                                  Object {
                                    "opacity": 0,
                                  },
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#999999",
                                      "fontSize": 25,
                                    },
                                    undefined,
                                    Object {
                                      "fontFamily": "Material Icons",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                              >
                                
                              </Text>
                            </View>
                          </View>
                          <Text
                            numberOfLines={1}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "textAlign": "center",
                                },
                                Object {
                                  "color": "#f4511e",
                                },
                                Object {
                                  "fontSize": 10,
                                },
                                Object {
                                  "fontSize": 12,
                                  "fontWeight": "bold",
                                },
                              ]
                            }
                          >
                            Home Page
                          </Text>
                        </View>
                        <View
                          accessibilityLabel="My Favourite(s), tab, 2 of 2"
                          accessibilityRole="button"
                          accessibilityState={
                            Object {
                              "selected": false,
                            }
                          }
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flex": 1,
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                              Object {
                                "flexDirection": "column",
                                "justifyContent": "flex-end",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "center",
                                    "height": "100%",
                                    "justifyContent": "center",
                                    "minWidth": 25,
                                    "position": "absolute",
                                    "width": "100%",
                                  },
                                  Object {
                                    "opacity": 0,
                                  },
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#f4511e",
                                      "fontSize": 35,
                                    },
                                    undefined,
                                    Object {
                                      "fontFamily": "Material Icons",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                              >
                                
                              </Text>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "center",
                                    "height": "100%",
                                    "justifyContent": "center",
                                    "minWidth": 25,
                                    "position": "absolute",
                                    "width": "100%",
                                  },
                                  Object {
                                    "opacity": 1,
                                  },
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#999999",
                                      "fontSize": 25,
                                    },
                                    undefined,
                                    Object {
                                      "fontFamily": "Material Icons",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                              >
                                
                              </Text>
                            </View>
                          </View>
                          <Text
                            numberOfLines={1}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "textAlign": "center",
                                },
                                Object {
                                  "color": "#8E8E8F",
                                },
                                Object {
                                  "fontSize": 10,
                                },
                                Object {
                                  "fontSize": 12,
                                  "fontWeight": "bold",
                                },
                              ]
                            }
                          >
                            My Favourite(s)
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
